### DISCUSSION TITLE
vscode插件生产的qt项目的compile_commands.json不包含头文件

### DISCUSSION THREAD

**Discussion by Mythos-404** [xmake-io/xmake#2839] [Q&A]
如题
但我看之前一些提的issues中说已经解决了
但使用 `xmake project -k compile_commands` 生成的会有头文件还有其他参数
![image](https://user-images.githubusercontent.com/96325829/190913470-7e3a7532-42a3-4a35-b3bb-b6ec06146fff.png)

用插件就没有
![image](https://user-images.githubusercontent.com/96325829/190913427-070686d6-20a8-4607-8997-69df3285fb3b.png)

还有一个小问题
就是调用 `xmake project -k compile_commands` 的时候回报这样的错误
![image](https://user-images.githubusercontent.com/96325829/190913370-52879400-ca48-4551-8303-96f71ed837b1.png)

**Comment by waruqi** [comment]
> 用插件就没有

这是插件的问题

> 还有一个小问题


这个 dev 版本之前已经修了，xmake update -s dev

**Reply by Mythos-404** [reply]
插件这个没法解决吗？

**Reply by waruqi** [reply]
没时间看，你先自己拉代码调下

**Reply by Mythos-404** [reply]
好吧

**Reply by Mythos-404** [reply]
![image](https://user-images.githubusercontent.com/96325829/190916852-d50a771b-28b2-4be1-a303-5178a89756f6.png)

我看了一下调用的这个文件

但我也没什么解决方法

我翻了一下issues中 https://github.com/xmake-io/xmake-vscode/issues/72#issuecomment-916804465 这位有一个临时解决方法

但您也说了有可能会导致bug

(虽然也能用)

**Reply by waruqi** [reply]
我修了 再试下

**Reply by Mythos-404** [reply]
可以了真的很感谢
